cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -20,6 +20,7 @@
 #include <memory>
 #include <string>
 #include <utility>
+#include <vector>
```

Comment:
nit: is this import necessary?

Reply:
Not for this PR, so I will revert. Nevertheless, I have noticed that there are several imports missing and probably some extra in several files. I think this should be its own JIRA issue.

Reply:
We have a tool called include-what-you-use, though I don't think it's been run a while. It might be good to give that a try again. (IIRC, it's a bit finicky to set up.)

Reply:
https://arrow.apache.org/docs/developers/cpp/development.html#cleaning-includes-with-include-what-you-use-iwyu